server/controllers/files.py

```diff
@@ -0,0 +1,57 @@
+import tempfile
+import logging
+
+import libcloud
+from flask import Blueprint, url_for, redirect, send_from_directory, abort
+from flask_login import login_required, current_user
+
+from server.extensions import storage
+from server.models import ExternalFile
+
+files = Blueprint('files', __name__)
+
+logger = logging.getLogger(__name__)
+
+@files.route("/files/<hashid:file_id>")
+@login_required
+def file_url(file_id):
+    ext_file = ExternalFile.query.filter_by(id=file_id, deleted=False).first()
+    if not ext_file or not ExternalFile.can(ext_file, current_user, 'download'):
+        return abort(404)
+    try:
+        storage_obj = ext_file.object
+    except libcloud.common.types.InvalidCredsError as e:
+        logger.warning("Could not get file {0} - {1}".format(file_id, ext_file.filename),
+                       exc_info=True)
+        storage_obj = None
```

Comment:
I don't think this is true anymore (the raise below it has been removed). (Though this line could be moved up to line 20)